[Payment Request] Update API to use inclusive language

aprotyas opened the pull request:
#### e404a58b8d31392246bc4045a77aab9b54fa9264
<pre>
[Payment Request] Update API to use inclusive language
<a href="https://bugs.webkit.org/show_bug.cgi?id=255359">https://bugs.webkit.org/show_bug.cgi?id=255359</a>
rdar://107259257

Reviewed by Aditya Keerthi.

It is preferred to avoid the enabled/disabled terminology in favour of available/unavailable.

* LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterAvailability.https-expected.txt: Added.
* LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterAvailability.https.html: Renamed from LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterMode.https.html.
* LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterMode.https-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/applepay/ApplePayFeature.h:
* Source/WebCore/Modules/applepay/ApplePayFeature.idl:
* Source/WebCore/Modules/applepay/ApplePayLaterAvailability.h: Renamed from Source/WebCore/Modules/applepay/ApplePayLaterMode.h.
* Source/WebCore/Modules/applepay/ApplePayLaterAvailability.idl: Renamed from Source/WebCore/Modules/applepay/ApplePayLaterMode.idl.
* Source/WebCore/Modules/applepay/ApplePayRequestBase.cpp:
(WebCore::convertAndValidate):
* Source/WebCore/Modules/applepay/ApplePayRequestBase.h:
* Source/WebCore/Modules/applepay/ApplePayRequestBase.idl:
* Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h:
(WebCore::ApplePaySessionPaymentRequest::applePayLaterAvailability const):
(WebCore::ApplePaySessionPaymentRequest::setApplePayLaterAvailability):
(WebCore::ApplePaySessionPaymentRequest::applePayLaterMode const): Deleted.
(WebCore::ApplePaySessionPaymentRequest::setApplePayLaterMode): Deleted.
* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/testing/MockPaymentCoordinator.cpp:
(WebCore::MockPaymentCoordinator::showPaymentUI):
* Source/WebCore/testing/MockPaymentCoordinator.h:
* Source/WebCore/testing/MockPaymentCoordinator.idl:
* Source/WebKit/Shared/ApplePay/cocoa/WebPaymentCoordinatorProxyCocoa.mm:
(WebKit::toPKApplePayLaterAvailability):
(WebKit::WebPaymentCoordinatorProxy::platformPaymentRequest):
(WebKit::toPKApplePayLaterMode): Deleted.
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.mm:
(IPC::ArgumentCoder&lt;WebCore::ApplePaySessionPaymentRequest&gt;::encode):
(IPC::ArgumentCoder&lt;WebCore::ApplePaySessionPaymentRequest&gt;::decode):

Canonical link: <a href="https://commits.webkit.org/263140@main">https://commits.webkit.org/263140@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/80090c0eb27c0be15ce3871ab283c2a3b67bca3c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [⏳ 🧪 style ](https://ews-build.webkit.org/#/builders/Style-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-16-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/WinCairo-EWS "Waiting in queue, processing has not started yet") 
| [⏳ 🧪 bindings ](https://ews-build.webkit.org/#/builders/Bindings-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitperl ](https://ews-build.webkit.org/#/builders/WebKitPerl-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitpy ](https://ews-build.webkit.org/#/builders/WebKitPy-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk1 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 gtk-wk2 ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-arm64-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-16-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 services ](https://ews-build.webkit.org/#/builders/Services-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-16-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2-stress ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/912 "Built successfully and passed tests") | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-9-Build-EWS "Waiting in queue, processing has not started yet") | | [⏳ 🛠 jsc-mips ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-9-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | [⏳ 🧪 jsc-mips-tests ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
```diff
@@ -357,6 +342,23 @@ typedef void(^PKCanMakePaymentsCompletion)(BOOL isValid, NSError *);
+#define PKApplePayLaterAvailable 0
+#define PKApplePayLaterUnavailableMerchantIneligible 1
+#define PKApplePayLaterUnavailableItemIneligible 2
+#define PKApplePayLaterUnavailableRecurringTransaction 3
```
Let's move these into `WebPaymentCoordinatorProxyCocoa.mm`. As we found out last week, it isn't safe to have `#define`s here, due to the import at the bottom of this file.

aprotyas replied:
Moved the staging code into `WebPaymentCoordinatorProxyCocoa.mm` (https://github.com/WebKit/WebKit/commit/fb9abe4c91ab2ea3951df9ec3b86783666cb9e45) 

Thoughts?

pxlcoder replied:
We only need to move the `#define`s.

aprotyas replied:
Yup. Please check https://github.com/WebKit/WebKit/commit/b33f3e1e9b538e099e3121818de80238325c2e73

aprotyas committed:
[Payment Request] Update API to use inclusive language
https://bugs.webkit.org/show_bug.cgi?id=255359
rdar://107259257

Reviewed by Aditya Keerthi.

It is preferred to avoid the enabled/disabled terminology in favour of available/unavailable.

* LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterAvailability.https-expected.txt: Added.
* LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterAvailability.https.html: Renamed from LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterMode.https.html.
* LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterMode.https-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/applepay/ApplePayFeature.h:
* Source/WebCore/Modules/applepay/ApplePayFeature.idl:
* Source/WebCore/Modules/applepay/ApplePayLaterAvailability.h: Renamed from Source/WebCore/Modules/applepay/ApplePayLaterMode.h.
* Source/WebCore/Modules/applepay/ApplePayLaterAvailability.idl: Renamed from Source/WebCore/Modules/applepay/ApplePayLaterMode.idl.
* Source/WebCore/Modules/applepay/ApplePayRequestBase.cpp:
(WebCore::convertAndValidate):
* Source/WebCore/Modules/applepay/ApplePayRequestBase.h:
* Source/WebCore/Modules/applepay/ApplePayRequestBase.idl:
* Source/WebCore/Modules/applepay/ApplePaySessionPaymentRequest.h:
(WebCore::ApplePaySessionPaymentRequest::applePayLaterAvailability const):
(WebCore::ApplePaySessionPaymentRequest::setApplePayLaterAvailability):
(WebCore::ApplePaySessionPaymentRequest::applePayLaterMode const): Deleted.
(WebCore::ApplePaySessionPaymentRequest::setApplePayLaterMode): Deleted.
* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/testing/MockPaymentCoordinator.cpp:
(WebCore::MockPaymentCoordinator::showPaymentUI):
* Source/WebCore/testing/MockPaymentCoordinator.h:
* Source/WebCore/testing/MockPaymentCoordinator.idl:
* Source/WebKit/Shared/ApplePay/cocoa/WebPaymentCoordinatorProxyCocoa.mm:
(WebKit::toPKApplePayLaterAvailability):
(WebKit::WebPaymentCoordinatorProxy::platformPaymentRequest):
(WebKit::toPKApplePayLaterMode): Deleted.
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.mm:
(IPC::ArgumentCoder<WebCore::ApplePaySessionPaymentRequest>::encode):
(IPC::ArgumentCoder<WebCore::ApplePaySessionPaymentRequest>::decode):

Canonical link: https://commits.webkit.org/263140@main
